### PR TITLE
[IMP] mail: sort messaging methods and make openDocument public

### PR DESCRIPTION
**Pull request by seb-odoo**
Also remove `forceOpenDocument` from `redirect` as it is strictly equivalent of
calling `openDocument`.

### CODE REVIEW

**Comment by seb-odoo** [comment]
@alexkuhn to review what we discussed extracted from my PR

File: addons/mail/static/src/messaging/entity/messaging/messaging.js [hunk 1]
```diff
@@ -32,23 +32,20 @@ function MessagingFactory({ Entity }) {
...
-            this.notificationHandler.start();
-            this.update({ isInitialized: true });
-        }
-
-        /**
-         * Stop messaging and related entities.
+         * Open the form view of the record with provided id and model.
+         *
+         * @param {integer} param1.id
+         * @param {string} param1.model
```
Comment (alexkuhn) [inline addons/mail/static/src/messaging/entity/messaging/messaging.js]: `param0`

**Review by alexkuhn — APPROVED** [review]
good otherwise

File: addons/mail/static/src/messaging/entity/messaging/messaging.js (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/src/messaging/entity/messaging/messaging.js]: Damn original code. I guess I'll remove param2 from redirect too while I'm at it.

Comment (seb-odoo) [reply addons/mail/static/src/messaging/entity/messaging/messaging.js]: All done, `* @param {Object} param0` was also missing here

**Comment by seb-odoo** [comment]
Green then merge.

**Pull request merged**